### DISCUSSION TITLE
How to access EDNS Option content received from dnsmasq

### DISCUSSION THREAD

**Discussion by MangeshBhamre** [PowerDNS/pdns#13891] [Q&A] [answered]
Hello team,

Here is my code snippet from dnsdist.conf. ecsOption.content is always nil. I wish to access cpe-id and mac-id  from EDNS Options.  Please help.
```lua
function getCpeID(dq)
   local cpeID
   if dq:getEDNSOptions() ~= nil then
        local ednsOptions = dq:getEDNSOptions()

        local ecsOption = ednsOptions[65073]
           if ecsOption then
                -- Assuming 'content' is the correct way to access the data you're interested in
                errlog("EDNS Option Code: " .. type(ecsOption)  .. tostring(ecsOption.content) )
                cpeID = tostring(ecsOption.content)
            end
    end
    return cpeID
end
```

**Comment by rgacogne** [comment, marked as answer]
I'm afraid you are incorrectly using the returned object, please have a look at https://dnsdist.org/reference/dq.html#ednsoptionview-object

**Reply by MangeshBhamre** [reply]
Thanks @rgacogne / Remi!

**Reply by rgacogne** [reply]
There is an example in our regression tests, if that helps: https://github.com/PowerDNS/pdns/blob/master/regression-tests.dnsdist/test_EDNSOptions.py#L7
